tools/integration_tests/gzip/helpers/helpers.go

```diff
@@ -131,7 +82,11 @@ func DownloadGzipGcsObjectAsCompressed(bucketName, objPathInBucket string) (stri
 		return "", fmt.Errorf("failed to get size of gcs object %s: %w", gcsObjectPath, err)
 	}
 
-	tempfile, err := CreateLocalTempFile(1, false)
+	content, err := CreateDataOfSize(1)
+	if err != nil {
+		return "", fmt.Errorf("failed to create data: %w", err)
+	}
+	tempfile, err := operations.CreateLocalTempFile(content, false)
```

Comment:
I see a disk-leak issue here. We create and write this file and return if the creation and writing succeed, but we don't delete this file if creation succeeds but the writing fails. Agree that this is a pre-existing issue, but we should fix it now that we have found it.